appservice/src/deploy/runWithZipStream.ts

```diff
@@ -16,11 +16,22 @@ import { localize } from '../localize';
 import { SiteClient } from '../SiteClient';
 import { getFileExtension } from '../utils/pathUtils';
 
-export async function getZipStream(context: IActionContext, fsPath: string, client: SiteClient): Promise<Readable> {
-    let zipStream: Readable;
+export async function runWithZipStream(context: IActionContext, fsPath: string, client: SiteClient, callback: (zipStream: Readable) => Promise<void>): Promise<void> {
```

Comment:
Is there any reason you couldn't just return the zip stream at the end and pass it in as an argument for `kuduClient.pushDeployment.zipPushDeploy` and `createWriteStreamToBlockBlob`?  Just never see us use the callback pattern.

Reply:
The main benefit of this pattern is I can reduce copied code. If I return zipStream, I would have to duplicate some of the logic below the comment "Setup several tasks" because of the ordering of the tasks and when they're awaited

I've used the callback pattern before. `registerCommand` and `callWithTelemetryAndErrorHandling` are the most common uses, but here's a few more examples:

https://github.com/microsoft/vscode-azuretools/blob/master/ui/index.d.ts#L224

https://github.com/microsoft/vscode-azurefunctions/blob/93a8ebe455785c9264d0b235df54468eaf567252/test/runWithSetting.ts#L16

Reply:
Okay fine, I will say that I feel like those use-cases are a little different, but I am okay with it

Reply:
![1](https://media.tenor.com/images/1104de76f163cf39b6da78565c7cd7da/tenor.gif)